task/amazon/resources/resource_security_group.go

```diff
@@ -0,0 +1,206 @@
+	case err == nil:
+		return nil
+	case err != universal.NotFoundError:
+		return err
+	}
+
+	createInput := ec2.CreateSecurityGroupInput{
+		GroupName:   aws.String(s.Identifier),
+		Description: aws.String(s.Identifier),
+		VpcId:       s.Dependencies.DefaultVPC.Resource.VpcId,
+		TagSpecifications: []types.TagSpecification{
+			{
+				ResourceType: types.ResourceTypeSecurityGroup,
+				Tags:         makeTagSlice(s.Identifier, s.Client.Tags),
+			},
+		},
+	}
+
+	group, err := s.Client.Services.EC2.CreateSecurityGroup(ctx, &createInput)
+	if err != nil {
+		return err
+	}
+
+	describeInput := ec2.DescribeSecurityGroupsInput{
+		GroupIds: []string{aws.ToString(group.GroupId)},
+	}
+
+	if err := ec2.NewSecurityGroupExistsWaiter(s.Client.Services.EC2).Wait(ctx, &describeInput, s.Client.Cloud.Timeouts.Create); err != nil {
+		return err
+	}
+
+	if err := s.Read(ctx); err != nil {
+		return err
+	}
+
+	// Revoke default full egress rule created for every new security group
+	revokeEgressInput := ec2.RevokeSecurityGroupEgressInput{
+		GroupId:       s.Resource.GroupId,
+		IpPermissions: s.generatePermissions(universal.FirewallRule{}),
+	}
+
+	if _, err := s.Client.Services.EC2.RevokeSecurityGroupEgress(ctx, &revokeEgressInput); err != nil {
+		return err
+	}
+
+	egressInput := ec2.AuthorizeSecurityGroupEgressInput{
+		GroupId:       s.Resource.GroupId,
+		IpPermissions: s.generatePermissions(s.Attributes.Egress),
+		TagSpecifications: []types.TagSpecification{
+			{
+				ResourceType: types.ResourceTypeSecurityGroupRule,
+				Tags:         makeTagSlice(s.Identifier, s.Client.Tags),
+			},
+		},
+	}
+
+	if _, err := s.Client.Services.EC2.AuthorizeSecurityGroupEgress(ctx, &egressInput); err != nil {
+		return err
+	}
+
+	ingressInput := ec2.AuthorizeSecurityGroupIngressInput{
+		GroupId:       s.Resource.GroupId,
+		IpPermissions: s.generatePermissions(s.Attributes.Ingress),
+		TagSpecifications: []types.TagSpecification{
+			{
+				ResourceType: types.ResourceTypeSecurityGroupRule,
+				Tags:         makeTagSlice(s.Identifier, s.Client.Tags),
+			},
+		},
+	}
+
+	if _, err := s.Client.Services.EC2.AuthorizeSecurityGroupIngress(ctx, &ingressInput); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *SecurityGroup) Read(ctx context.Context) error {
+	input := ec2.DescribeSecurityGroupsInput{
+		Filters: []types.Filter{
+			{
+				Name:   aws.String("tag:Name"),
+				Values: []string{s.Identifier},
+			},
+		},
+	}
+
+	securityGroups, err := s.Client.Services.EC2.DescribeSecurityGroups(ctx, &input)
+	if err != nil {
+		return err
+	}
+
+	if len(securityGroups.SecurityGroups) < 1 {
+		return universal.NotFoundError
+	}
+
+	s.Resource = &securityGroups.SecurityGroups[0]
+	return nil
+}
+
+func (s *SecurityGroup) Update(ctx context.Context) error {
+	return universal.NotImplementedError
+}
+
+func (s *SecurityGroup) Delete(ctx context.Context) error {
+	switch err := s.Read(ctx); {
+	case err == universal.NotFoundError:
```

Comment:
This is not _that_ suspicious: AWS Security Groups are the only resource that requires a `Read` operation to determine whether it exists or not, because they are being referenced by name and not by ARN.

Reply:
what I say is return nil if `universal.NotFoundError:` looks weird

Reply:
See above: https://github.com/iterative/terraform-provider-iterative/pull/237#discussion_r746966778

Reply:
If you're trying to `Delete` and it doesn't exist, the operation has been succesful, right? 😄